src/rdkafka_aux.c

```diff
@@ -97,3 +97,76 @@ void rd_kafka_topic_result_destroy (rd_kafka_topic_result_t *terr) {
 void rd_kafka_topic_result_free (void *ptr) {
         rd_kafka_topic_result_destroy((rd_kafka_topic_result_t *)ptr);
 }
+
+rd_kafka_resp_err_t
+rd_kafka_group_result_error (const rd_kafka_group_result_t *groupres) {
+        return groupres->err;
+}
+
+const char *
+rd_kafka_group_result_error_string (const rd_kafka_group_result_t *groupres) {
+        return groupres->errstr;
+}
+
+const char *
+rd_kafka_group_result_name (const rd_kafka_group_result_t *groupres) {
+        return groupres->group;
+}
+
+rd_kafka_group_result_t *
+rd_kafka_group_result_copy (const rd_kafka_group_result_t *groupres) {
+        return rd_kafka_group_result_new(groupres->group, -1, groupres->err, groupres->errstr);
+}
+
+/**
+ * @brief Create new group_result (single allocation).
+ *
+ * @param group Group string, if group_size is != -1 it does not have to
+ *              be nul-terminated.
+ * @param group_size Size of group, or -1 to perform automatic strlen()
+ * @param err Error code
+ * @param errstr Optional error string.
+ *
+ * All input arguments are copied.
+ */
+
+rd_kafka_group_result_t *
+rd_kafka_group_result_new (const char *group, ssize_t group_size,
+                           rd_kafka_resp_err_t err,
+                           const char *errstr) {
+        size_t tlen = group_size != -1 ? (size_t)group_size : strlen(group);
+        size_t elen = errstr ? strlen(errstr) + 1 : 0;
+        rd_kafka_group_result_t *terr;
+
+        terr = rd_malloc(sizeof(*terr) + tlen + 1 + elen);
```

Comment:
For non-fast-path objects like this I prefer to use rd_calloc() to avoid missing an initialization if the object is ever extended.